### PR TITLE
GT-1300 support animations in about banner

### PR DESCRIPTION
**Pull request by levieggertcru**
This PR adds support for animations in the ToolDetailView top banner.  Added an enum ToolDetailBanner that the view observes on the viewModel to control if the banner should display an image, youtube video, or animation.
Honor Restored and The Four both support animations in the tool details.

### CODE REVIEW

File: godtools/App/Services/RealmDatabase/RealmDatabase.swift [hunk 1]
```diff
@@ -12,7 +12,7 @@ import RealmSwift
 class RealmDatabase {
     
     private static let config: Realm.Configuration = RealmDatabase.createConfig
-    private static let schemaVersion: UInt64 = 5
+    private static let schemaVersion: UInt64 = 6
```
Comment (frett) [inline godtools/App/Services/RealmDatabase/RealmDatabase.swift:15]: out of curiosity does Realm on iOS automatically generate migrations? or does just bumping the schemaVersion reset the database completely?

Comment (frett) [reply godtools/App/Services/RealmDatabase/RealmDatabase.swift:15]: from: https://www.mongodb.com/docs/realm/sdk/swift/examples/modify-an-object-schema/

> If your schema update adds optional properties or removes properties, Realm Database can perform the migration automatically. You only need to increment the schemaVersion.

That's nice

Comment (levieggertcru) [reply godtools/App/Services/RealmDatabase/RealmDatabase.swift:15]: Yeah they do some things automatically.

**Review by rachaelblue — DISMISSED** [review]
Animations are cool!!

File: godtools/App/Features/ToolDetails/Models/ToolDetailBanner.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+//  ToolDetailBanner.swift
+//  godtools
+//
+//  Created by Levi Eggert on 5/9/22.
+//  Copyright © 2022 Cru. All rights reserved.
+//
+
+import UIKit
+
+enum ToolDetailBanner {
```
[the remark this answers is not in the document]
Comment (levieggertcru) [reply godtools/App/Features/ToolDetails/Models/ToolDetailBanner.swift:11]: Hey @rachaelblue one thing to note is that although the view's inherit from UIViewController, we're not using them as ViewControllers from an architectural point of view as in ```MVC```.  They're simply Views according to ```MVVM```.     
The ```ToolDetailBanner``` I'm not sure on a best naming for that one.  ViewModels are a view representation so they aren't supposed to be providing actual Views, just the data that a view renders.  (String, Image, etc.). What are your thoughts on naming for special data types as in this one?

[the remark this answers is not in the document]
Comment (levieggertcru) [reply godtools/App/Features/ToolDetails/Models/ToolDetailBanner.swift:11]: Would be cleaner to create a separate View for the ToolDetailBannerView that encapsulates the imageView, animationView, videoView, etc and then provide a viewModel for it.

[the remark this answers is not in the document]
Comment (rachaelblue) [reply godtools/App/Features/ToolDetails/Models/ToolDetailBanner.swift:11]: Oh ok gotcha, that's interesting about view/VC.  So I guess that is the naming standard-- everything should be one of MVVM? 

And yeah I'd be inclined to name the enum `ToolDetailBannerType` or something similar.  But you're right, would maybe fit the pattern better to make it a separate view and viewModel.

[the remark this answers is not in the document]
Comment (levieggertcru) [reply godtools/App/Features/ToolDetails/Models/ToolDetailBanner.swift:11]: Hey @rachaelblue UIKit was written with ```MVC``` in mind, where you'd write both view logic and controller logic in your UIViewController classes.  We'd even write navigation logic in there at times.  
In GodTools even though we inherit UIViewController, we aren't using that paradigm, just the view part.  With the switch to SwiftUI our Views will make a lot more sense.  No more need to subclass UIViewController.

uikit-and-mvc

[the remark this answers is not in the document]
Comment (levieggertcru) [reply godtools/App/Features/ToolDetails/Models/ToolDetailBanner.swift:11]: Going to make that change to ToolDetailBannerType.

**Pull request merged**